### ISSUE TITLE
Proposal to Enhance Testing Infrastructure and Automate Workflows

### ISSUE THREAD

**Issue by Aglag257** [CatimaLoyalty/Android#1848]
Hello,

As a first-time contributor, I've noticed an opportunity to strengthen the repo's testing framework (which also aligns with one of the goals that is written in the documentation) . I propose we create a dedicated GitHub project to systematically track and improve our testing efforts. This project could include:

    1) Mapping out Missing Test Classes: Identifying areas of the codebase that are currently under-tested.
    2) Increasing Test Coverage: Setting goals for and measuring the test coverage to ensure more robust code.
    3) Implementing Testing Automation Workflows: Automating testing processes to streamline development and improve efficiency.

I believe these enhancements will not only boost our testing efficiency but also contribute to a more stable and reliable codebase. I'm eager to discuss this further and contribute to its implementation.

Thank you for considering this suggestion.

**Comment by TheLastProject** [comment]
I don't think we need a new GitHub project. And I'm not interested in "setting goals" as I'm a single hobbyist just writing this as a hobby. So "code coverage percentages" are not a big concern to me.

However, more tests are generally welcome as they tend to simplify testing during development, so feel free to suggest/provide possible test cases. Catima uses Robolectric, so more Robolectric tests are probably easiest. But it may be worth considering if whatever is the current standard for Android tests will work too :)

For new automation workflows, #712 is still open, having screenshot generation and/or tests would help too.

Given this is not really an issue, I'm converting this to a discussion :)
